### PR TITLE
Aeroglass features

### PR DESCRIPTION
**Pull request by Matyooo**
fixed an invalid character in the iOS C++ source which causes a compile error with the generated java code in Android Studio under Windows 10

added Android Studio and Gradle specific things to .gitignore

added a feature to turn off FPS reduction for the android widget because we do the FPS reduction in our own code

### CODE REVIEW

File: iOS/G3MiOSSDK/Commons/Rendererers/BusyMeshRenderer.cpp [hunk 1]
```diff
@@ -71,9 +71,9 @@ Mesh* BusyMeshRenderer::createMesh(const G3MRenderContext* rc) {
   const int viewPortWidth  = camera->getViewPortWidth();
   const int viewPortHeight = camera->getViewPortHeight();
   const int minSize = (viewPortWidth < viewPortHeight) ? viewPortWidth : viewPortHeight;
-  const float outerRadius = minSize / 15.0f;
-
-  const IMathUtils* mu = IMathUtils::instance();
+  const float outerRadius = minSize / 18.0f;
```
Comment (DiegoGomezDeck) [inline iOS/G3MiOSSDK/Commons/Rendererers/BusyMeshRenderer.cpp]: this change was a test for the translator, isn't it?

File: iOS/G3MiOSSDK/Commons/Rendererers/BusyMeshRenderer.cpp [hunk 2]
```diff
@@ -71,9 +71,9 @@ Mesh* BusyMeshRenderer::createMesh(const G3MRenderContext* rc) {
...
+    ILogger::instance()->logError("This a test log line");
```
Comment (DiegoGomezDeck) [inline iOS/G3MiOSSDK/Commons/Rendererers/BusyMeshRenderer.cpp]: ditto

Comment (DrakkLord) [inline iOS/G3MiOSSDK/Commons/Rendererers/BusyMeshRenderer.cpp]: Yes these were only added to test the translator.

Comment (DiegoGomezDeck) [inline iOS/G3MiOSSDK/Commons/Rendererers/BusyMeshRenderer.cpp]: Hi @DrakkLord 

Can you, please, roll it back to the values in purgatory?   After that, I think we can merge this PR.

**Comment by DrakkLord** [comment]
Hi @DiegoGomezDeck !

I have reverted those test changes to what is on purgatory now.
Also in the meantime we merged additional features into this branch, in the future we will not do this, we will use feature branches and issue pull requests on them without adding any new commits until the request accepted/rejected.

Summary of features added by new commit
- added a feature for android specific code to allow storing large files within the downlaod cache by saving them next to the database file, this is due to a problem coming from android's SQLite which crashes around 1-2Mb of blob size, this is a workaround by storing a the file name reference instead and saving the file externally
- added a new GL feature which allows using custom opengl shader names and allows setting uniforms and attributes from that feature, also added code that makes sure that G3M does not try to set unrecognized attributes and/or uniforms

**Pull request closed without merging**